ingestion/src/metadata/data_insight/processor/web_analytic_report_data_processor.py

```diff
@@ -107,12 +108,27 @@ def _refine_entity_event(self) -> Generator[dict, WebAnalyticEventData, None]:
                 continue
 
             entity_obj = EntityObj(split_url[0], split_url[1])
+            entity_type = entity_obj.entity_type
+            re_pattern = re.compile(f"(.*{entity_type}/{entity_obj.fqn})")
+
+            if (
+                entity_obj.fqn in refined_data
+                and not refined_data[entity_obj.fqn]["entityHref"]
+            ):
+                # if we've seen the entity previously but were not able to get
+                # the URL we'll try again from the new event.
+                try:
+                    entity_href = re.search(re_pattern, event.eventData.fullUrl).group(
```

Comment:
Is this local variable used? I think we are overriding it anyway on line 163.

Reply:
@nahuelverdugo it is a different logic here
```
               # if we've seen the entity previously but were not able to get
                # the URL we'll try again from the new event.
```

Reply:
Then.. where are we using `entity_href`? Should we assign it to the object?
```
refined_data[split_url[1]]["entityHref"] = entity_href
```